app/components/common/userInput/index.tsx

```diff
@@ -17,12 +31,14 @@ const UserInput: React.FC<UserInputProps> = ({ label, placeholder, link }) => (
         </button>
       )}
       <input
-        type="text"
-        data-testid="user-input"
+        type={type ?? 'text'}
+        data-testid={dataTestId ?? 'user-input'}
         className={` bg-stone-50 text-sm p-3  focus:outline-none ${
           link ? 'basis-3/4 border-l-0 rounded-r-lg ' : 'rounded-lg w-full'
         }   border-solid border  border-stone-400`}
         placeholder={placeholder}
+        value={value}
+        onChange={(e) => setValue(e.target.value)}
```

Comment:
What is the type of `e` here ?

Reply:
> What is the type of `e` here ?

as this is an `onChange` event of an input element, by default the type is `React.ChangeEvent<HTMLInputElement>` so I did not think the need to be specific here. Do you want me to still specify the type?